lib/galaxy/webapps/galaxy/api/workflows.py

```diff
@@ -117,33 +117,41 @@ def get_workflows_list( self, trans, for_menu, kwd ):
         missing_tools = util.string_as_bool( kwd.get( 'missing_tools', 'False' ) )
         rval = []
         filter1 = ( trans.app.model.StoredWorkflow.user == trans.user )
+        user = trans.get_user()
         if show_published:
             filter1 = or_( filter1, ( trans.app.model.StoredWorkflow.published == true() ) )
         for wf in trans.sa_session.query( trans.app.model.StoredWorkflow ).filter(
                 filter1, trans.app.model.StoredWorkflow.table.c.deleted == false() ).order_by(
                 desc( trans.app.model.StoredWorkflow.table.c.update_time ) ).all():
-            if for_menu:
-                item = wf.to_dict()
-            else:
-                item = wf.to_dict( value_mapper={ 'id': trans.security.encode_id } )
-                encoded_id = trans.security.encode_id(wf.id)
-                item['url'] = url_for('workflow', id=encoded_id)
+
+            item = wf.to_dict( value_mapper={ 'id': trans.security.encode_id } )
+            encoded_id = trans.security.encode_id(wf.id)
+            item['url'] = url_for('workflow', id=encoded_id)
             item['owner'] = wf.user.username
             item['number_of_steps'] = len( wf.latest_workflow.steps )
+            item['show_in_tool_panel'] = False
+            item['wf_id'] = wf.id
```

Comment:
`get_workflow_menu` api is nowhere used and it used to display the unencoded workflow id. But the workflows which are present in the tools menu already have a boolean flag `show_in_tool_panel` in the list of workflows itself. But, PUT api is used to save the list using the encoded workflow ids. That's why I removed this Get API. There is already a separate api to list workflows. It was not required anymore.